omp/solver/idr_kernels.cpp

```diff
@@ -191,10 +195,13 @@ void initialize(std::shared_ptr<const OmpExecutor> exec, const size_type nrhs,
         }
 
         auto norm = zero<remove_complex<ValueType>>();
-#pragma omp parallel for reduction(+ : norm)
-        for (size_type j = 0; j < num_cols; j++) {
-            norm += squared_norm(subspace_vectors->at(row, j));
-        }
+        run_kernel_reduction(
+            exec,
+            [](auto col, auto row, auto subspace_vectors) {
+                return squared_norm(subspace_vectors(row, col));
+            },
+            GKO_KERNEL_REDUCE_SUM(remove_complex<ValueType>), &norm, num_cols,
+            static_cast<int64>(row), subspace_vectors);
```

Comment:
Is it for fix or optimization?

Reply:
It's a first step towards unifying it, but also a workaround for NVHPC's limited `pragma omp reduction` feature set

Reply:
Do you have any specific feature name for the unsupported one and the corresponding error message?

Reply:
basically `pragma omp (parallel) for reduction` is supported, but `omp declare reduction` is not

Reply:
I see. because omp `run_kernel_reduction` does not use omp reduction (declare for complex), it works via `run_kernel_reduction`